compositor_render/src/renderer/node.rs

```diff
@@ -184,17 +179,42 @@ impl Node {
     }
 }
 
-#[derive(Debug, thiserror::Error)]
-pub enum CreateNodeError {
-    #[error("Shader \"{0}\" does not exist. You have to register it first before using it in the scene definition.")]
-    ShaderNotFound(RendererId),
-
-    #[error("Error while validating the parameters for the shader node:\n{0}")]
-    ShaderNodeParametersValidationError(#[from] ParametersValidationError),
-
-    #[error("Instance of web renderer \"{0}\" does not exist. You have to register it first before using it in the scene definition.")]
-    WebRendererNotFound(RendererId),
+pub(crate) trait NodeSpecExt {
+    fn constraints<'a>(
+        &self,
+        renderers: &'a Renderers,
+    ) -> Result<&'a NodeConstraints, UpdateSceneError>;
+}
 
-    #[error("Image \"{0}\" does not exist. You have to register it first before using it in the scene definition.")]
-    ImageNotFound(RendererId),
+impl NodeSpecExt for NodeSpec {
+    fn constraints<'a>(
+        &self,
+        renderers: &'a Renderers,
+    ) -> Result<&'a NodeConstraints, UpdateSceneError> {
+        match &self.params {
+            NodeParams::WebRenderer { instance_id } => renderers
+                .web_renderers
+                .get_ref(instance_id)
+                .map(|web_renderer| web_renderer.constraints())
+                .ok_or_else(|| {
+                    UpdateSceneError::CreateNodeError(
```

Comment:
I did not want to create separate error types just for that, and this is exactly error message that user should get if it happened